Replace deprecated calls from aws-sdk-go-v2

**Describe the bug**
We still use some deprecated funcs, see here:
https://github.com/NBISweden/sda-cli/actions/runs/22570648330/job/65377326635?pr=657
```
  Error: list/list_test.go:89:14: SA1019: manager.NewUploader is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 (staticcheck)
  	uploader := manager.NewUploader(s3Client)
  	            ^
  Error: list/list_test.go:93:15: SA1019: uploader.Upload is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 (staticcheck)
  	if _, err := uploader.Upload(context.Background(), &s3.PutObjectInput{
  	             ^
  Error: upload/upload.go:133:14: SA1019: manager.NewUploader is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 (staticcheck)
  	uploader := manager.NewUploader(s3Client)
  	            ^
  Error: upload/upload.go:224:18: SA1019: uploader.Upload is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 (staticcheck)
  		result, err := uploader.Upload(ctx, &s3.PutObjectInput{
  		               ^
  Error: upload/upload.go:229:14: SA1019: manager.Uploader is deprecated: superceded by feature/s3/transfermanager. See https://github.com/aws/aws-sdk-go-v2/discussions/3306 (staticcheck)
  		}, func(u *manager.Uploader) {
  		           ^
  5 issues:
  * staticcheck: 5
  
  Error: issues found
``` 
**Steps to reproduce**
Run linter as above
**Expected behavior**
- [ ] clean tests
- [ ] Tests verifying the fix are added

**Additional context**

**Estimation of size**: small/medium/big

**Estimation of priority**: low/medium/high

--- a/.github/integration/tests/10_encrypt_decrypt.sh
+++ b/.github/integration/tests/10_encrypt_decrypt.sh
@@ -6,6 +6,9 @@ source "$test_dir/../scripts/checkers.sh"
 # Create random file
 dd if=/dev/urandom of=data_file count=10 bs=1M
 
+# Larger file to exercise multipart upload path (>50 MiB chunk threshold)
+dd if=/dev/urandom of=data_file_big count=60 bs=1M
+
 # Create key pair
 if ( yes "" | ./sda-cli createKey sda_key ) ; then
     echo "Created key pair for encryption"
@@ -18,6 +21,9 @@ fi
 ./sda-cli encrypt --key sda_key.pub.pem data_file
 
 check_encrypted_file data_file.c4gh
+
+./sda-cli encrypt --key sda_key.pub.pem data_file_big
+check_encrypted_file data_file_big.c4gh
 
 
 # Create folder and encrypt files in it

--- a/.github/integration/tests/20_upload.sh
+++ b/.github/integration/tests/20_upload.sh
@@ -19,6 +19,20 @@ cp data_files_enc/data_file.c4gh data_files_enc/dir1/dir2/data_file2.c4gh
 ./sda-cli --config testing/s3cmd.conf upload data_file.c4gh
 check_uploaded_file "test/$user/data_file.c4gh" data_file.c4gh
 
+# Exercise multipart code path (PartSizeBytes=50MiB, file>60MiB encrypted)
+./sda-cli --config testing/s3cmd.conf upload data_file_big.c4gh
+if ! s3cmd -c testing/directS3 ls "s3://test/$user/data_file_big.c4gh" | grep -q data_file_big.c4gh; then
+    echo "Multipart upload missing from s3 backend"
+    exit 1
+fi
+s3size=$(s3cmd -c testing/directS3 ls "s3://test/$user/data_file_big.c4gh" | awk '{print $3}')
+local_size=$(wc -c < data_file_big.c4gh | awk '{print $1}')
+if [ "$s3size" != "$local_size" ]; then
+    echo "Multipart upload size mismatch: s3=$s3size local=$local_size"
+    exit 1
+fi
+echo "Multipart upload verified ($s3size bytes)"
+
 # Upload the file twice check that this returns an error
 msg=$(./sda-cli --config testing/s3cmd.conf upload data_file.c4gh 2>&1 | tail -1 || true)
 if ! grep -q "Error:" <<< "$msg"

--- a/go.mod
+++ b/go.mod
@@ -6,8 +6,8 @@ require (
 	github.com/aws/aws-sdk-go-v2 v1.41.6
 	github.com/aws/aws-sdk-go-v2/config v1.32.16
 	github.com/aws/aws-sdk-go-v2/credentials v1.19.15
-	github.com/aws/aws-sdk-go-v2/feature/s3/manager v1.22.15
-	github.com/aws/aws-sdk-go-v2/service/s3 v1.99.1
+	github.com/aws/aws-sdk-go-v2/feature/s3/transfermanager v0.1.18
+	github.com/aws/aws-sdk-go-v2/service/s3 v1.100.0
 	github.com/dustin/go-humanize v1.0.1
 	github.com/golang-jwt/jwt v3.2.2+incompatible
 	github.com/hashicorp/go-version v1.9.0

--- a/list/list_test.go
+++ b/list/list_test.go
@@ -18,7 +18,7 @@ import (
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials"
-	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/transfermanager"
 	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/golang-jwt/jwt"
 	"github.com/johannesboyne/gofakes3"
@@ -86,11 +86,11 @@ func (s *ListTestSuite) SetupSuite() {
 	if err != nil {
 		s.FailNow("failed to create s3 bucket", err)
 	}
-	uploader := manager.NewUploader(s3Client)
+	uploader := transfermanager.New(s3Client)
 
 	fileToUpload := strings.NewReader("test content")
 	s.testFilePath = "dummy/testfile"
-	if _, err := uploader.Upload(context.Background(), &s3.PutObjectInput{
+	if _, err := uploader.UploadObject(context.Background(), &transfermanager.UploadObjectInput{
 		Body:            fileToUpload,
 		Bucket:          aws.String("dummy"),
 		Key:             aws.String(s.testFilePath),

--- a/upload/upload.go
+++ b/upload/upload.go
@@ -5,6 +5,7 @@ import (
 	"encoding/hex"
 	"errors"
 	"fmt"
+	"net/url"
 	"os"
 	"path"
 	"path/filepath"
@@ -276,10 +277,14 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 		}
 
 		p.Shutdown()
+		keySegments := strings.Split(path.Join(targetDir, outFiles[k]), "/")
+		for i := range keySegments {
+			keySegments[i] = url.PathEscape(keySegments[i])
+		}
 		fmt.Printf("file uploaded to %s/%s/%s\n",
 			strings.TrimRight(config.HostBase, "/"),
-			config.AccessKey,
-			path.Join(targetDir, outFiles[k]))
+			url.PathEscape(config.AccessKey),
+			strings.Join(keySegments, "/"))
 	}
 
 	return nil

--- a/upload/upload.go
+++ b/upload/upload.go
@@ -16,7 +16,7 @@ import (
 	"github.com/aws/aws-sdk-go-v2/aws"
 	awsconfig "github.com/aws/aws-sdk-go-v2/config"
 	"github.com/aws/aws-sdk-go-v2/credentials"
-	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/transfermanager"
 	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/neicnordic/crypt4gh/keys"
 	"github.com/spf13/cobra"
@@ -130,7 +130,9 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 	})
 
 	// Create an uploader with the session and default options
-	uploader := manager.NewUploader(s3Client)
+	uploader := transfermanager.New(s3Client, func(o *transfermanager.Options) {
+		o.PartSizeBytes = config.MultipartChunkSizeMb * 1024 * 1024
+	})
 	for k, filename := range files {
 		// create progress bar instance
 		p := mpb.New()
@@ -221,15 +223,11 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 		)
 
 		// Upload the file to S3.
-		result, err := uploader.Upload(ctx, &s3.PutObjectInput{
+		result, err := uploader.UploadObject(ctx, &transfermanager.UploadObjectInput{
 			Body:            bar.ProxyReader(fs.Reader),
 			Bucket:          aws.String(config.AccessKey),
 			Key:             aws.String(path.Join(targetDir, outFiles[k])),
 			ContentEncoding: aws.String(config.Encoding),
-		}, func(u *manager.Uploader) {
-			u.PartSize = config.MultipartChunkSizeMb * 1024 * 1024
-			// Delete parts of failed multipart, since we cannot currently continue them
-			u.LeavePartsOnError = false
 		})
 		// Print the progress bar. Second check is to filter out some junk from the output
 		if result != nil && result.VersionID != nil {
@@ -278,7 +276,10 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 		}
 
 		p.Shutdown()
-		fmt.Printf("file uploaded to %s\n", aws.ToString(&result.Location))
+		fmt.Printf("file uploaded to %s/%s/%s\n",
+			strings.TrimRight(config.HostBase, "/"),
+			config.AccessKey,
+			path.Join(targetDir, outFiles[k]))
 	}
 
 	return nil

